### PR TITLE
fix(datepicker): set datepicker touched if bluring input or closing the pane

### LINKED ISSUE #6598: md-datepicker doesn't support $touched in a form
md-datepicker should be fully supported by forms. Using the control doesn't play nicely with angular form controls like $touched.

http://codepen.io/anon/pen/XXRVmp

### PR DESCRIPTION
**Pull request by DevVersion**
Fixes #6598

### CODE REVIEW

**Comment by ThomasBurleson** [comment]
@jelbourn - can you review?

File: src/components/datepicker/datePicker.js [hunk 1]
```diff
@@ -598,6 +600,7 @@
    * @param {boolean} isFocused
    */
   DatePickerCtrl.prototype.setFocused = function(isFocused) {
+    if (isFocused === false) this.ngModelCtrl.$setTouched();
```
Comment (jelbourn) [inline src/components/datepicker/datePicker.js]: Re-style to 
```js
if (!isFocused) {
  this.ngModelCtrl.$setTouched();
}
```

**Comment by jelbourn** [comment]
LGTM aside from one style comment. Thanks for the fix!

**Comment by DevVersion** [comment]
@jelbourn Updated :)

**Pull request closed without merging**